crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -447,3 +470,43 @@ pub fn annotated_assignment_default_in_stub(
     }
     checker.diagnostics.push(diagnostic);
 }
+
+// PYI052
+pub fn assignment_default_in_async_stub(
+    checker: &mut Checker,
+    targets: &[Expr],
+    value: &Expr,
+    annotation: Option<&Expr>,
+) {
+    if is_valid_default_value_without_annotation(value) {
+        return;
+    }
+
+    if let Some(annotation) = annotation {
+        if is_valid_default_value_with_annotation(value, checker, true) {
+            // Annoying special-casing to exclude enums from Y052
+            // Check if you're inside a class and if the class is not an enum class
+            let inside_non_enum_class = is_inside_class(&checker.ctx, &targets[0]) && !is_inside_enum_class(&checker.ctx, &targets[0]);
+
+            if inside_non_enum_class {
+                let mut diagnostic = Diagnostic::new(AssignmentDefaultInStub, value.range()); // Make sure AssignmentDefaultInAsyncStub is defined in your code
```

Comment:
We need to define a new diagnostic type to represent Y052 itself: `UnannotatedAssignmentInStub`. Can you look at how `AssignmentDefaultInStub` is defined and define a new diagnostic for it?

Reply:
Yeah I added it to the `codes.rs` and `registry.rs`